src/test/java/org/broadinstitute/hellbender/utils/read/SamAssertionUtils.java

```diff
@@ -17,30 +17,46 @@
-             final SamReader reader2 = SamReaderFactory.makeDefault().validationStringency(validationStringency).open(sam2)) {
+    private static SamReader getReader(File sam, ValidationStringency validationStringency, File reference) {
+        return SamReaderFactory.makeDefault().validationStringency(validationStringency).referenceSequence(reference).open(sam);
+    }
+
+    public static void assertSamsEqual(final File sam1, final File sam2, ValidationStringency validationStringency, final File reference) throws IOException {
+        try (final SamReader reader1 = getReader(sam1, validationStringency, reference);
+             final SamReader reader2 = getReader(sam2, validationStringency, reference)) {
             final SamComparison comparison = new SamComparison(reader1, reader2);
             final boolean equal = comparison.areEqual();
             Assert.assertTrue(equal, "SAM file output differs from expected output");
         }
     }
 
+    public static void assertSamsEqual(final File sam1, final File sam2, ValidationStringency validationStringency) throws IOException {
+        assertSamsEqual(sam1, sam2, validationStringency, null);
+    }
+
+    public static void assertSamsEqual(final File sam1, final File sam2, final File reference) throws IOException {
+        assertSamsEqual(sam1, sam2, ValidationStringency.DEFAULT_STRINGENCY, reference);
+    }
+
     public static void assertSamsEqual(final File sam1, final File sam2) throws IOException {
-        assertSamsEqual(sam1, sam2, ValidationStringency.DEFAULT_STRINGENCY);
+        assertSamsEqual(sam1, sam2, ValidationStringency.DEFAULT_STRINGENCY, null);
     }
 
-    public static void assertSamsNonEqual(final File sam1, final File sam2) throws IOException {
-        try (final SamReader reader1 = SamReaderFactory.makeDefault().open(sam1);
-             final SamReader reader2 = SamReaderFactory.makeDefault().open(sam2)) {
+    public static void assertSamsNonEqual(final File sam1, final File sam2, final File reference) throws IOException {
+        try (final SamReader reader1 = getReader(sam1, ValidationStringency.DEFAULT_STRINGENCY, reference);
+             final SamReader reader2 = getReader(sam2, ValidationStringency.DEFAULT_STRINGENCY, reference)) {
             final SamComparison comparison = new SamComparison(reader1, reader2);
             final boolean equal = comparison.areEqual();
             Assert.assertFalse(equal, "SAM files are expected to differ, but they do not");
         }
     }
 
-    public static void assertSamValid(final File sam) throws IOException {
-        try (final SamReader samReader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.LENIENT).open(sam)) {
+    public static void assertSamsNonEqual(final File sam1, final File sam2) throws IOException {
+        assertSamsNonEqual(sam1,sam2,null);
+    }
+
+    public static void assertSamValid(final File sam, final File reference) throws IOException {
```

Comment:
Provide a 3-argument overload of `assertSamValid()` that takes a sam, reference, and `ValidationStringency`, and make the others call into it.